app/data_model/progress_store.py

```diff
@@ -15,96 +16,145 @@ class ProgressStore:
     that have been started.
     """
 
-    def __init__(self, in_progress_sections: Mapping = None) -> None:
+    def __init__(self, in_progress_sections: List[MutableMapping] = None) -> None:
         """
         Instantiate a ProgressStore object that tracks the status of sections and its completed locations
         Args:
             in_progress_sections: A hierarchical dict containing the section status and completed locations
         """
         self._is_dirty = False  # type: bool
-        self._progress = {}  # type: MutableMapping
+        self._progress = self._build_map(
+            in_progress_sections or []
+        )  # type: MutableMapping
 
-        if not in_progress_sections:
-            return
+    def __contains__(self, section_key) -> bool:
+        return section_key in self._progress
 
-        for section_id, serialised_section in in_progress_sections.items():
-            self._progress[section_id] = {
-                'status': serialised_section['status'],
+    @staticmethod
+    def _build_map(in_progress_sections: List[MutableMapping]) -> MutableMapping:
+        """
+        Builds the progress_store's data structure from a list of progress dictionaries
+
+        The `section_key` is tuple consisting of `section_id` and the `list_item_id`
+        The `section_progress` is a mapping created from the Progress object
+
+        Example structure:
+        {
+            ('some-section', 'a-list-item-id'): {
+                'section_id': 'some-section',
+                'status': 'COMPLETED',
+                'list_item_id': 'a-list-item-id',
                 'locations': [
-                    Location.from_dict(location_dict=location)
-                    for location in serialised_section['locations']
+                    {
+                        'section_id': 'some-section',
+                        'block_id': 'some-block',
+                        'list_name': 'people',
+                        'list_item_id': 'a-list-item-id',
+                    }
                 ],
             }
+        }
+        """
 
-    def __contains__(self, section_id):
-        return section_id in self._progress
+        return {
+            (section['section_id'], section.get('list_item_id')): Progress.from_dict(
+                section
+            )
+            for section in in_progress_sections
```

Comment:
As above, are these sections? Could they be named better is not?

Reply:
As discussed, this is now `section_progress in section_progress_list`